app/javascript/components/mentoring/AddDiscussionPost.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback } from 'react'
+import { usePanel } from '../../hooks/use-panel'
+import { DiscussionPostForm } from './DiscussionPostForm'
+
+export const AddDiscussionPost = ({
```

Comment:
I personally still think top-level function declarations are easier to read, but do whatever :P